File: 7kyu/mumbling.js
/*
This time no story, no theory. The examples below show you how to write function accum:
accum("abcd") -> "A-Bb-Ccc-Dddd"
accum("cwAt") -> "C-Ww-Aaa-Tttt"
*/

function accum(str) {

    let arr = str.toLowerCase().split('')
    let newArr = [];

    for (let i = 0; i < arr.length; i++) {
        let subString = "";
        
        subString += arr[i].toUpperCase()
        subString+= arr[i].repeat(i+1).slice(0,-1); // delete last char in a string
        newArr.push(subString)
    }
    return newArr.join('-')
}

console.log(accum("mumbling"))

File: 7kyu/mumbling.py
"""
Input > Output:
mumbling("abcd") -> "A-Bb-Ccc-Dddd"
mumbling("cwAt") -> "C-Ww-Aaa-Tttt"

1. turn all letters to lowercase and create a letters array;
2. iterate over the letters array and use * operator on each letter:
- arr[i] - accesses the letter in the array
- (i + 1) sets the number of repetition needed starting from 1

"""

# Solution 1

def mumbling(st):

    newArr = []
    arr = [l.lower() for l in st]

    for i, l in enumerate(arr):

        newArr.append(arr[i] * (i + 1))

    return "-".join(newArr).title()


print(mumbling("mum"))


# Solution 2

# def mumbling(st):
#     newArr = []
#     arr = [l.lower() for l in st]

#     for idx, letter in enumerate(arr):
#         subString = ""
#         subString += arr[idx].upper()
#         subString += (arr[idx] * (idx + 1))[0:-1]
#         newArr.append(subString)
#     return "-".join(newArr)
